### ISSUE TITLE
fast-logger: Breaking API change in minor version bump

### ISSUE THREAD

**Issue by snoyberg** [kazu-yamamoto/logger#194]
3.0.3:

```haskell
newFileLoggerSet :: BufSize -> FilePath -> IO LoggerSet
```

3.0.4

```haskell
newFileLoggerSet :: BufSize -> Maybe Int -> FilePath -> IO LoggerSet
```

Discovered with a Stackage build, where downstream packages (tinylog and logging) were broken.

**Comment by kazu-yamamoto** [comment]
I categorized that `System.Log.FastLogger.LoggerSet` is a kind of internal modules.
But if this is a real problem, I will recover the backward compatibility soon.

**Comment by snoyberg** [comment]
For now, I've opened up downstream issues:

* https://github.com/jwiegley/logging/issues/25
* https://gitlab.com/twittner/tinylog/-/issues/4

I've also blocked fast-logger-3.0.4 from Stackage until downstream packages resolve the issue. There's no immediate problem for Stackage, so it's your decision on whether to make a change for the downstream packages.

**Comment by kazu-yamamoto** [comment]
Fixed in v3.0.5. 3.0.4 is deprecated.

**Comment by snoyberg** [comment]
Awesome, thanks!
